src/main/kotlin/no/nav/familie/ef/mottak/service/SøknadService.kt

```diff
@@ -158,4 +159,25 @@ class SøknadService(
     fun oppdaterSøknad(søknad: Søknad) {
         søknadRepository.update(søknad)
     }
+
+    @Transactional
+    fun reduserSøknad(søknadId: String) {
+        val søknad = søknadRepository.findByIdOrNull(søknadId) ?: return
+        if (søknad.journalpostId == null) {
+            throw IllegalStateException("Søknad $søknadId er ikke journalført og kan ikke reduseres.")
+        }
+        vedleggRepository.deleteBySøknadId(søknadId)
+        dokumentasjonsbehovRepository.deleteById(søknadId)
+        val t = søknad.copy(søknadPdf = null)
+        søknadRepository.update(t)
+    }
+
+    @Transactional
+    fun slettSøknad(søknadId: String) {
+        val søknad = søknadRepository.findByIdOrNull(søknadId) ?: return
+        if (søknad.journalpostId == null) {
+            throw IllegalStateException("Søknad $søknadId er ikke journalført og kan ikke slettes.")
+        }
+        søknadRepository.deleteById(søknadId)
```

Comment:
Hvis noen går inn på en link på mitt-nav, med link til søknaden så kommer de få en feil.
Er det ønskelig? 
```kotlin
val søknad: Søknad =
            søknadRepository.findByIdOrNull(søknadId.toString()) ?: throw ApiFeil(
                "Fant ikke søknad for id $søknadId",
                HttpStatus.BAD_REQUEST
            )
```

Mulig dette burde sjekkes med funksjonelle. Gjetter på at det er tilstrekkelig å kanskje bare returnere noe annet enn feil, men det må sikkert håndteres både i frontend og backend

Reply:
Fjerner sletting av søknadspdf. Det er ikke den som står for mest plass uansett.

Reply:
Etter å ha sjekket ut - det finnes ikke noen "link" til søknaden etter at den er sendt inn? Kun til ettersending?

Reply:
Det er sletting av søknaden som blir "problemet", ikke pdfen

Reply:
Du tenker på de gamle linkene som vi sluttet å bruke for lenge siden? De tror jeg dessverre ikke gir brukeren noe verdi i dag heller. 

Men siden i fjor høst har vi kun lenket direkte til ettersending _uten_ noe id fra søknaden.